docs/developing.rst

```diff
@@ -571,6 +571,25 @@ Example::
 
             toil.exportFile(outputFileID, 'file:///some/other/local/path')
 
+
+There are some trade-offs to file staging that should be considered. It may be more efficient to
+directly download a file for cases in which the cost of the copy operation imposed by an import is
+unnecessary. For example, a direct download is better for large files that are used only once.
+Importing can be more efficient in cases where the file in question is needed several times. For
+example, Amazon's S3 platform, used in the AWS job store, is better optimized for reads and writes
+to buckets from an EC2 instance in the same region. Thus an import is justified if a
+source file is in a different region than that of the job store and the workflow is running on
+AWS EC2 instances. Note that `toil-scripts/lib`_ contains common functionality for downloading URLs in cases
+where a download is preferred to file staging.
+
+.. _toil-scripts/lib: https://github.com/BD2KGenomics/toil-scripts/blob/master/src/toil_scripts/lib/urls.py
+
+
+Files that are accessed exactly once during a workflow, should be imported on workers. This is done
+inside job functions with calls to the :meth:`toil.job.Job.FileStore.importFile` method. Importing
+files on a worker where possible will allow for performance gains due to the parallelization of
+concurrently running jobs as well as the added benefit of retries on import failures.
```

Comment:
Still not happy with this. I think we should mention the different cases first and then list the pros and cons for each one. The cases are: 

1. Import
    
    1a. Import on leader
    
    1b. Import on worker

2. Direct download on worker

Note how 1a and 1b are subitems of 1. They share certain pros and cons.

Be sure to mention that files local to the leader must be imported on the leader.